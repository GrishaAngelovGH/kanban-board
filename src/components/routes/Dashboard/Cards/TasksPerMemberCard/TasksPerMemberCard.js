import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  BarElement,
  Title,
  Tooltip,
  Legend
} from "chart.js"

import { Bar } from "react-chartjs-2"
import Card from "../Card"

import userRepository from "persistent/persistentUserRepository"
import boardRepository from "persistent/persistentKanbanBoardRepository"

ChartJS.register(
  CategoryScale,
  LinearScale,
  BarElement,
  Title,
  Tooltip,
  Legend
)

const users = userRepository.getUsers()

const labels = users.map(v => v.name)

const chartData = users.map(v => boardRepository.getColumnsWithAssignedTasksForUser(v.id).reduce((a, b) => a + b.items.length, 0))

const options = {
  responsive: true,
  plugins: {
    legend: {
      position: "top"
    }
  }
}

const data = {
  labels,
  datasets: [
    {
      label: "Tasks Per Users",
      data: chartData,
      backgroundColor: "#80b3ce",
    }
  ],
}

const TasksPerMemberCard = () => (
  <Card title="Tasks Per Users Card">
    <Bar options={options} data={data} />
  </Card>
)

export default TasksPerMemberCard