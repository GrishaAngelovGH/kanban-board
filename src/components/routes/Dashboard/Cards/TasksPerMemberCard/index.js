import TasksPerMemberCard from "./TasksPerMemberCard"

export default TasksPerMemberCard