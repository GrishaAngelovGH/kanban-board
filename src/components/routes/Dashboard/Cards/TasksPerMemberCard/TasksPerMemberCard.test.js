import { render } from "@testing-library/react"

import TasksPerMemberCard from "./TasksPerMemberCard"

jest.mock('react-chartjs-2', () => ({ Bar: () => <div>Bar Chart</div> }))

test("should render TasksPerMemberCard route component", () => {
  const view = render(
    <TasksPerMemberCard />
  )

  expect(view).toMatchSnapshot()
})