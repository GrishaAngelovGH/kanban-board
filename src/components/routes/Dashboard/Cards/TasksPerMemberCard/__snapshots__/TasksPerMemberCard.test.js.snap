// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TasksPerMemberCard route component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="col col-md-5 bg-white rounded shadow"
        style="height: 350px;"
      >
        <div
          class="d-flex justify-content-between align-items-center"
        >
          <p
            class="m-0 ms-2"
          >
            Tasks Per Users Card
          </p>
          <div
            class="d-flex align-items-center"
          >
            <i
              class="bi bi-star fs-4 text-warning"
            />
            <i
              class="bi bi-x fs-2"
            />
          </div>
        </div>
        <div
          class="d-flex justify-content-center"
        >
          <div>
            Bar Chart
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="col col-md-5 bg-white rounded shadow"
      style="height: 350px;"
    >
      <div
        class="d-flex justify-content-between align-items-center"
      >
        <p
          class="m-0 ms-2"
        >
          Tasks Per Users Card
        </p>
        <div
          class="d-flex align-items-center"
        >
          <i
            class="bi bi-star fs-4 text-warning"
          />
          <i
            class="bi bi-x fs-2"
          />
        </div>
      </div>
      <div
        class="d-flex justify-content-center"
      >
        <div>
          Bar Chart
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
